### ISSUE TITLE
ORCM orun still tries to execute binary for misconfigured MPI job 

### ISSUE THREAD

**Issue by noahv** [open-mpi/orcm#129]
If we try and run a job for a user that does not exist on the CNs (which is detected) then the job is still attempted to be run:

<pre>
 /opt/open-rcm/bin/orun -n 1 a.out                                                                                                  
[[604,0],0]:  session: 11 allocated for the job 
--------------------------------------------------------------------------
Open MPI was unable to obtain the username in order to create a path
for its required temporary directories.  This type of error is usually
caused by a transient failure of network-based authentication services
(e.g., LDAP or NIS failure due to network congestion), but can also be
an indication of system misconfiguration.
 
Please consult your system administrator about these issues and try
again.
--------------------------------------------------------------------------
[c1:48502] *** Process received signal ***
[c1:48502] Signal: Segmentation fault (11)
[c1:48502] Signal code: Address not mapped (1)
[c1:48502] Failing at address: 0x30
[c1:48502] [ 0] /lib64/libpthread.so.0(+0xf890)[0x7f7706429890]
[c1:48502] [ 1] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/openmpi/mca_btl_openib.so(+0x1f3c5)[0x7f7701a6b3c5]
[c1:48502] [ 2] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/openmpi/mca_btl_openib.so(+0x20583)[0x7f7701a6c583]
[c1:48502] [ 3] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/openmpi/mca_btl_openib.so(ompi_btl_openib_connect_base_select_for_local_port+0xcf)[0x7f7701a6a07f]
[c1:48502] [ 4] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/openmpi/mca_btl_openib.so(+0xf81b)[0x7f7701a5b81b]
[c1:48502] [ 5] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/libmpi.so.1(mca_btl_base_select+0x163)[0x7f77066ba193]
[c1:48502] [ 6] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/openmpi/mca_bml_r2.so(mca_bml_r2_component_init+0x11)[0x7f7701c7c931]
[c1:48502] [ 7] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/libmpi.so.1(mca_bml_base_init+0x6e)[0x7f77066b995e]
[c1:48502] [ 8] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/openmpi/mca_pml_ob1.so(+0x5ecf)[0x7f76fffe3ecf]
[c1:48502] [ 9] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/libmpi.so.1(mca_pml_base_select+0x399)[0x7f77066cc269]
[c1:48502] [10] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/libmpi.so.1(ompi_mpi_init+0x4c7)[0x7f770667f377]
[c1:48502] [11] /opt/fsp/pub/mpi/openmpi-gnu-fsp/1.8.3/lib/libmpi.so.1(MPI_Init+0x16b)[0x7f770669e98b]
[c1:48502] [12] a.out[0x40095e]
[c1:48502] [13] /lib64/libc.so.6(__libc_start_main+0xf5)[0x7f7706093b05]
[c1:48502] [14] a.out[0x400869]
[c1:48502] *** End of error message ***
orun_recv - JOB_COMPLETE
--------------------------------------------------------------------------
orun noticed that process rank 0 with PID 0 on node c1 exited on signal 11 (Segmentation fault).
--------------------------------------------------------------------------
</pre>

**Comment by vpedabal** [comment]
We need to fix *ALL* references to "Open MPI" within ORCM.

**Comment by noahv** [comment]
Hi Rajesh,

I created a new user, validation_user, on the head node only.

I am seeing that we are able to run the hostname command on the CN even though the validation_user only exists on the HN.

I would expect that instead we would give the error where ORCM is unable to obtain the username (as originally seen) but then not attempt to run the binary.

Removed the fixed label for this issue.

<pre>
validation_user:~> whoami
validation_user
validation_user:~> /opt/open-rcm/bin/orun -n 1 hostname                                                                                            
[[6229,0],0]:  session: 13 allocated for the job 
c1
orun_recv - JOB_COMPLETE
validation_user:~> /opt/open-rcm/bin/orun -n 1 whoami                                                                                              
[[6217,0],0]:  session: 14 allocated for the job 
whoami: cannot find name for user ID 1003
orun_recv - JOB_COMPLETE
--------------------------------------------------------------------------
orun detected that one or more processes exited with non-zero status, thus causing
the job to be terminated. The first process to do so was:
 
  Process name: [[14,1],0]
  Exit code:    1
--------------------------------------------------------------------------
</pre>
